proto_generator/protogenerator.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Binary proto_generator generates Protobuf3 code corresponding to an input
+// YANG schema. The input set of modules are read, parsed using goyang, and
+// handled as input to the ygen package which generates the corresponding
+// set of Protobuf3 messages.
+package main
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/golang/glog"
+	"github.com/openconfig/goyang/pkg/yang"
+	"github.com/openconfig/ygot/ygen"
+)
+
+var (
+	yangPaths        = flag.String("path", "", "Comma separated list of paths to be recursively searched for included modules or submodules within the defined YANG modules.")
+	compressPaths    = flag.Bool("compress_paths", false, "If set to true, the schema's paths are compressed, according to OpenConfig YANG module conventions.")
+	excludeModules   = flag.String("exclude_modules", "", "Comma separated set of module names that should be excluded from code generation this can be used to ensure overlapping namespaces can be ignored.")
+	packageName      = flag.String("package_name", "openconfig", "The name of the Proto package that generated messages should belong to as their parent.")
+	enumPackageName  = flag.String("enum_package_name", "enums", "The name of the package within the generated package that should contain global enum definitions.")
+	outputDir        = flag.String("output_dir", "", "The path to which files should be output, hierarchical folders are created for the generated messages.")
+	ignoreCircDeps   = flag.Bool("ignore_circdeps", false, "If set to true, circular dependencies between submodules are ignored.")
+	generateFakeRoot = flag.Bool("generate_fakeroot", false, "If set to true, a fake element at the root of the data tree is generated. By default the fake root entity is named Device, its name can be controlled with the fakeroot_name flag.")
+	fakeRootName     = flag.String("fakeroot_name", "", "The name of the fake root entity.")
```

Comment:
Add "Device" default? Then you can trim the generateFakeRoot comment.

Reply:
Done.